packages/smb/src/components/ui/negotiators-list/negotiator-office-selectbox.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react'
+import ReactDataSheet from 'react-datasheet'
+import { Cell } from '@reapit/elements/src/components/Spreadsheet/types'
+import { OfficesQueryResponse } from '../offices-tab/offices-tab'
+import { prepareCreateNegeotiatorParams } from './negotiators-list'
+
+export type NegotiatorOfficeSelectboxProps = {
+  cellRenderProps: ReactDataSheet.CellRendererProps<Cell, string | null>
+  officeData?: OfficesQueryResponse
+  spreadsheetData: Cell[][]
+  createNegotiator: (params) => void
+}
+
+export const handleOnChange = (officeData, setValue, spreadsheetData, row, createNegotiator) => {
+  return e => {
+    const selectedOfficeId = e.target.value
+    const selectedOffice = officeData?.GetOffices?._embedded?.find(office => office.id === selectedOfficeId)
+    setValue(selectedOfficeId)
+    const createNegotiatorVariables = {
+      ...prepareCreateNegeotiatorParams(spreadsheetData, row),
+      officeId: selectedOfficeId,
+    }
+    createNegotiator({
+      variables: createNegotiatorVariables,
+      optimisticResponse: {
+        __typename: 'Mutation',
+        CreateNegotiator: {
+          ...createNegotiatorVariables,
+          id: Math.random(),
+          created: new Date().toISOString,
+          modified: new Date().toISOString,
+          workPhone: null,
+          metadata: null,
+          _eTag: null,
+          _links: null,
+          _embedded: {
+            office: {
+              id: selectedOfficeId,
+              name: selectedOffice?.name,
+            },
+          },
+          __typename: 'NegotiatorModel',
+        },
+      },
+    })
+  }
+}
+
+export const NegotiatorOfficeSelectbox: React.FC<NegotiatorOfficeSelectboxProps> = ({
+  cellRenderProps,
+  officeData,
+  spreadsheetData,
+  createNegotiator,
+}) => {
+  const {
+    row,
+    cell: { value: office },
+  } = cellRenderProps
+
+  const [value, setValue] = React.useState<string>()
+
+  const officeId = office?.split('|')[0]
+  const officeName = office?.split('|')[1]
+  const selectedRow = spreadsheetData[row]
+  const id = selectedRow[6].value
```

Comment:
what is [6]

Reply:
Hi @tanphamhaiduong ,
"6" is the column index of ID in Negotiator Spreadsheet Row which was defined when creating DataTableRow 
```
const dataRows: DataTableRow[][] = negotiators.map((negotiator: NegotiatorModel) => {
    const { name, jobTitle, email, mobilePhone, active, id, _eTag, _embedded } = negotiator
    const office: OfficeModel = _embedded?.office
    return [
      { value: name },
      { value: jobTitle },
      { value: email },
      { value: mobilePhone },
      { readOnly: true, value: office?.name, CustomComponent: OfficeSelectbox },
      { readOnly: true, value: active, CustomComponent: StatusCheckbox },
      { value: id, className: 'hidden' }, // 6th index
      { value: _eTag, className: 'hidden' },
    ]
  })
```